src/main/java/de/dennisguse/opentracks/TrackDetailActivity.java

```diff
@@ -387,9 +428,13 @@ private void handleIntent(Intent intent) {
                 return;
             }
             trackId = waypoint.getTrackId();
+            if (trackId == -1L) {
+                finish();
+                return;
+            }
         }
+
         if (trackId == -1L) {
-            finish();
```

Comment:
Why was finish() removed?
The activity should not start if it got an invalid parameter, right?

Reply:
That `finish()` was there because TrackDetailActivity needed it to show or to record a Track. Then TrackDetailActivity always needed the trackId.

Now, if TrackDetailActivity is started without trackId is because is started to record a new Track, so in `handleIntent` it doesn't finish, because TrackDetailActivity will start the service and will got trackId by its own.

Look at this (in onResume())
```
if (trackId != -1L) {
    trackDataHub.loadTrack(trackId);
    trackController.onResume(isRecording(), recordingTrackPaused);
} else {
    startRecording();
}
```

If there isn't trackId in EXTRA_TRACK_ID then it calls to startRecording. Otherwise it shows the track.

So, we could say that TrackDetailActivity handle three "modes":
- With trackId it enters in mode show.
- Without trackId (trackId == -1) it enters in mode recording.
- With trackId if user click on record button it enters in mode resuming.

Reply:
Thanks for the clarification!
Then let's merge it ;)